libraries/lib-ffmpeg-support/wrappers/AVCodecContextWrapper.cpp

```diff
@@ -66,8 +68,18 @@ const AVCodecContext* AVCodecContextWrapper::GetWrappedValue() const noexcept
 
 AVCodecContextWrapper::~AVCodecContextWrapper()
 {
-   if (mIsOwned && mFFmpeg != nullptr && mAVCodecContext != nullptr && mFFmpeg->avcodec_free_context != nullptr)
+   // avcodec_free_context, complementary to avcodec_alloc_context3, is
+   // not necessarily loaded
+   if (mIsOwned && mFFmpeg != nullptr && mAVCodecContext != nullptr && mFFmpeg->avcodec_free_context != nullptr) {
+      // FFmpeg docs aren't quite clear, but I think an explicit
```

Comment:
https://ffmpeg.org/doxygen/trunk/group__lavc__core.html#gaf4daa92361efb3523ef5afeb0b54077f

>Do not use this function. Use avcodec_free_context() to destroy a codec context (either open or closed). Opening and closing a codec context multiple times is not supported anymore – use multiple codec contexts instead.

Reply:
Okay, it looked to me as if some resource reclamation done in the previous code was lost in this rewrite.

But it should be asked, does the documentation you cite apply to all versions of FFmpeg that we will support, or only to the latest?

If only the latest, then there may be some work to do to get it right for both.  Maybe then one .inl file won't be enough to compile the correct overrides for AVCodecContextWrapperImpl

Reply:
I took the best of both worlds in the end :-)

If `avcodec_free_context` is unavailable - I close the codec. The corresponding code will be pushed soon